### PR TITLE
[ONNX] Adjust `is_train` flag for onnx pass deduplicate initializers

### PR DESCRIPTION
**Pull request by BowenBao**
Previous logic didn't consider the case for TrainingMode.PRESERVE.
A more direct way is to check `model.training`, which is the accurate
training mode, set by `exporter_context(model, training)`.

### CODE REVIEW

**Review by garymm — APPROVED** [review]
😹

**Comment by BowenBao** [comment]
I will fix the CI issue before merging

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -818,8 +818,8 @@ def _export(model, args, f, export_params=True, verbose=False, training=None,
                 # NOTE: cannot call DCE after this pass. DCE will remove function definition nodes.
                 node_attr_to_name = torch._C._jit_pass_onnx_function_extraction(
                     graph, export_modules_as_functions, list(params_dict.keys()))
-            params_dict = torch._C._jit_pass_onnx_deduplicate_initializers(graph, params_dict,
-                                                                           training == TrainingMode.TRAINING)
+            params_dict = torch._C._jit_pass_onnx_deduplicate_initializers(
+                graph, params_dict, hasattr(model, "training") and model.training)
```
Comment (garymm) [inline torch/onnx/utils.py:822]: `getattr(model, "training", False)`

**Pull request closed without merging**